### PR TITLE
test: Move tests to streaming consumer package

### PR DESCRIPTION
**Pull request by lynnagara**
This moves the tests related to the streaming Kafka consumer out of
Snuba tests and into the correct pacakge.

### CODE REVIEW

**Comment by lynnagara** [comment]
I'm a bit stumped as to why mypy is complaining

**Review by evanh — APPROVED** [review]
It's complaining now because the CI skipped the `tests` folder for typing. Since you moved it into this package it's now typing the test files. I think you still have to put `# type: ignore` on test decorators.

**Comment by lynnagara** [comment]
@evanh I did intend to start typing these test files since these should all pass (and do when I run this locally). It only complains about the fixture, but there doesn't seem to be anything invalid about that from what I can tell.

**Comment by evanh** [comment]
Maybe CI is using the wrong version of mypy/pytest? That error about untyped decorators still shows up for me locally.

**Comment by lynnagara** [comment]
Now to figure out why pytest is failing (incorrectly)

**Review by evanh — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -39,6 +39,7 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           pip install mypy==0.800
+          pip install pytest==6.1.2
```
Comment (fpacifici) [inline .github/workflows/ci.yml:42]: Couldn't we just upgrade the dependency in our requirements.txt ?

File: streaming_kafka_consumer/tests/backends/test_kafka.py [hunk 2]
```diff
@@ -54,7 +52,9 @@ def test_payload_pickle_out_of_band() -> None:
 
 class KafkaStreamsTestCase(StreamsTestMixin[KafkaPayload], TestCase):
 
-    configuration = get_default_kafka_configuration()
+    configuration = build_kafka_configuration_with_overrides(
+        {"bootstrap.servers": os.environ.get("DEFAULT_BROKERS", "localhost:9092")}
+    )
```
Comment (fpacifici) [inline streaming_kafka_consumer/tests/backends/test_kafka.py:57]: Why was this needed ?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (lynnagara) [reply .github/workflows/ci.yml:42]: This job does not install any other packages from requirements

File: streaming_kafka_consumer/tests/backends/test_kafka.py (hunk 2 above)
Comment (lynnagara) [reply streaming_kafka_consumer/tests/backends/test_kafka.py:57]: This test requires the actual Kafka connection (and we obviously don't want to depend on Snuba settings to get that). This will probably be tweaked (and moved to a central place) later when this moves to another repository as that environment variable may not be used anymore.

**Review by fpacifici — APPROVED** [review]
